visualiser/graph.py

```diff
@@ -0,0 +1,102 @@
+import copy
+
+from visualiser import Node, Edge
+
+
+class Graph:
+    def __init__(self, name='', **attrs):
+        self._name = name
+        self._attrs = attrs
+
+        self._nodes = []
+        self._edges = []
+
+    def get_node(self, name):
+        filtered_nodes = list(filter(lambda node: node.name == name, self._nodes))
+
+        return filtered_nodes[0] if len(filtered_nodes) else None
+
+    def remove_node(self, _node):
+        if isinstance(_node, Node):
+            self._nodes = list(filter(lambda node: node.name != _node.name, self._nodes))
+            return
+
+        self._nodes = list(filter(lambda node: node.name != _node, self._nodes))
+
+    def add_node(self, node):
+        if self.get_node(node.name) is not None:
+            self.remove_node(node)
+
+        self._nodes.append(copy.deepcopy(node))
+
+    def set_node_attributes(self, node, **attrs):
+        for key, value in attrs.items():
+            self.set_node_attribute(node, key, value)
```

Comment:
**Suggestion:** Consider adding error handling for when the node or edge is not of the correct type. This will prevent the program from crashing when an invalid type is passed.
```suggestion
        def add_node(self, node):
                if not isinstance(node, Node):
                    raise TypeError('node must be of type Node')
                if self.get_node(node.name) is not None:
                    self.remove_node(node)
                self._nodes.append(copy.deepcopy(node))

        def add_edge(self, edge):
                if not isinstance(edge, Edge):
                    raise TypeError('edge must be of type Edge')
                if self.get_edge(edge.name) is not None:
                    self.remove_edge(edge)
                self._edges.append(copy.deepcopy(edge))
```